hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +279,230 @@ default Table newHmsTable(String hmsTableOwner) {
+
+    } catch (LockException e) {
+      throw new CommitFailedException(e);
+    } finally {
+      cleanupMetadataAndUnlock(lock, io, commitStatus, newMetadataLocation);
+    }
+  }
+
+  default String fullName() {
+    return catalogName() + "." + database() + "." + table();
+  }
+
+  default StorageDescriptor storageDescriptor(BaseMetadata metadata, boolean hiveEngineEnabled) {
+    return HiveOperationsBase.storageDescriptor(
+        metadata.schema(), metadata.location(), hiveEngineEnabled);
+  }
+
+  default Set<String> obsoleteProps(Configuration conf, BaseMetadata base, BaseMetadata metadata) {
+    Set<String> obsoleteProps = Sets.newHashSet();
+    if (base != null) {
+      obsoleteProps =
+          base.properties().keySet().stream()
+              .filter(key -> !metadata.properties().containsKey(key))
+              .collect(Collectors.toSet());
+    }
+
+    if (!conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false)) {
+      obsoleteProps.add(StatsSetupConst.COLUMN_STATS_ACCURATE);
+    }
+
+    return obsoleteProps;
+  }
+
+  /**
+   * Returns if the hive engine related values should be enabled on the table, or not.
+   *
+   * <p>The decision is made like this:
+   *
+   * <ol>
+   *   <li>Table property value {@link TableProperties#ENGINE_HIVE_ENABLED}
+   *   <li>If the table property is not set then check the hive-site.xml property value {@link
+   *       ConfigProperties#ENGINE_HIVE_ENABLED}
+   *   <li>If none of the above is enabled then use the default value {@link
+   *       TableProperties#ENGINE_HIVE_ENABLED_DEFAULT}
+   * </ol>
+   *
+   * @param metadata Table metadata to use
+   * @return if the hive engine related values should be enabled or not
+   */
+  default boolean hiveEngineEnabled(Configuration conf, BaseMetadata metadata) {
+    if (metadata.properties().get(TableProperties.ENGINE_HIVE_ENABLED) != null) {
+      // We know that the property is set, so default value will not be used,
+      return metadata.propertyAsBoolean(TableProperties.ENGINE_HIVE_ENABLED, false);
+    }
+
+    return conf.getBoolean(
+        ConfigProperties.ENGINE_HIVE_ENABLED, TableProperties.ENGINE_HIVE_ENABLED_DEFAULT);
+  }
+
+  /**
+   * Returns if the hive locking should be enabled on the table, or not.
+   *
+   * <p>The decision is made like this:
+   *
+   * <ol>
+   *   <li>Table property value {@link TableProperties#HIVE_LOCK_ENABLED}
+   *   <li>If the table property is not set then check the hive-site.xml property value {@link
+   *       ConfigProperties#LOCK_HIVE_ENABLED}
+   *   <li>If none of the above is enabled then use the default value {@link
+   *       TableProperties#HIVE_LOCK_ENABLED_DEFAULT}
+   * </ol>
+   *
+   * @param metadata Table metadata to use
+   * @return if the hive engine related values should be enabled or not
+   */
+  default boolean hiveLockEnabled(Configuration conf, BaseMetadata metadata) {
+    if (metadata.properties().get(TableProperties.HIVE_LOCK_ENABLED) != null) {
```

Comment:
`engine.hive.lock.enabed` will available for Views as per @pvary recommendation [here](https://github.com/apache/iceberg/pull/9852#discussion_r1519220818)

Reply:
@nk1506 I understand that `engine.hive.lock.enabed` will be available for views. My point is that there's no reason to check `if (metadata.properties().get(TableProperties.HIVE_LOCK_ENABLED) != null)` on a view, because a view won't have `TableProperties.HIVE_LOCK_ENABLED` set in its properties